olive/systems/azureml/aml_pass_runner.py

```diff
@@ -100,8 +91,8 @@ def main(raw_args=None):
     model_json = output_model.to_json()
 
     # Replace output model HF config with input model HF config
-    if hf_config:
-        model_json["config"]["hf_config"] = hf_config
+    if input_model_config["config"].get("hf_config"):
+        model_json["config"]["hf_config"] = input_model_config["config"]["hf_config"]
```

Comment:
We don't use snpe or openvino models on aml so this will not fail but this assumes all models have `hf_config`. 

Is this part required? I think the hf_config needs to be provided to the output model by the pass that created it. Otherwise we have mismatch in the output model's config between aml system and local system. Here the system is autoinserting hf_config to all output models.

Reply:
I assume all previous model which has `hf_config` is a hf model so the next output model is also a hf model. This is correct?

Reply:
Currently, the hf configs are propagated only from pytorch model to composite onnx model by the onnx conversion pass. 

None of the other passes or models use it.

Reply:
But I think propagating to all output models sounds fine too. We would need to update the output of all the passes to do this though.

Reply:
If we can do this when running pass that would be great, then we can remove this logic here, otherwise the output hf onnx model will not have the hf config data.

Reply:
Maybe we can do it in a different PR so that your PR doesn't need to worry about it?
I also think it is better for the passes to do it if we want all models to have hf_config. This way the behavior is consistent for all systems since the pass inserts it.

Reply:
Yeah let's do this in a separate PR. Thanks!